pkg/dyninst/decode/decoder.go

```diff
@@ -244,12 +226,12 @@ func (d *Decoder) encodeValue(
 		if err != nil {
 			return err
 		}
-		addrString := fmt.Sprintf("0x%x", header.Address)
+		addrString := fmt.Sprintf("0x%x", address)
 		err = enc.WriteToken(jsontext.String(addrString))
 		if err != nil {
 			return err
 		}
-		if _, ok := currentlyEncoding[key]; !ok && !dontParseValue {
+		if _, alreadyEncoding := currentlyEncoding[key]; !alreadyEncoding && ok {
```

Comment:
I think we should be doing rather different things for the already encoding thing and the case where we didn't find the data item. If we didn't find it, it's probably because we hit the depth limits and then I think we should consider something like `"notCapturedReason": "depth"` -- though we'll want to at some point differentiate between depth limits and failing to dereference a pointer. In side-eye we did some tracking of failures to read memory. For now let's assume it's depth.

If we hit a cycle then we should say that in the `"notCapturedReason"`